flowmachine/flowmachine/core/cache.py

```diff
@@ -239,6 +246,47 @@ def reset_cache(connection: "Connection") -> None:
             trans.execute(f"DROP TABLE IF EXISTS cache.{table[0]} CASCADE")
         trans.execute("TRUNCATE cache.cached CASCADE")
         trans.execute("TRUNCATE cache.dependencies CASCADE")
+    resync_redis_with_cache(connection=connection, redis=redis)
+
+
+def resync_redis_with_cache(connection: "Connection", redis: StrictRedis) -> None:
+    """
+    Reset redis to be in sync with the current contents of the cache.
+
+    Parameters
+    ----------
+    connection : Connection
+    redis : StrictRedis
+
+    Returns
+    -------
+    None
+
+    Notes
+    -----
+    You _must_ ensure that no queries are currently running when calling this function.
+    Any queries currently running will no longer be tracked by redis, and UNDEFINED BEHAVIOUR
+    will occur.
+    """
+    logger.debug("Redis resync")
+    qry = f"SELECT query_id FROM cache.cached"
+    queries_in_cache = connection.fetch(qry)
+    logger.debug("Redis resync", queries_in_cache=queries_in_cache)
+    redis.flushdb()
+    logger.debug("Flushing redis.")
+    for event in (QueryEvent.QUEUE, QueryEvent.EXECUTE, QueryEvent.FINISH):
```

Comment:
Not sure if it matters, but is it worth swapping the order of the two for loops, so that the logging messages for each `query_id` are closer together as it's propelled through the different states? Might make the logging output a tiny bit more readable. (Although again, might not matter much because if everything goes well then we're not interested in the logging output and if something goes wrong it'll throw an error anyway.)

Reply:
My thinking was that if something goes wrong, you'd have everything in the same state which seemed slightly nicer to me.

Reply:
👍